auth0/resource_auth0_role.go

```diff
@@ -0,0 +1,167 @@
+package auth0
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"gopkg.in/auth0.v1"
+	"gopkg.in/auth0.v1/management"
+)
+
+func newRole() *schema.Resource {
+	return &schema.Resource{
+
+		Create: createRole,
+		Update: updateRole,
+		Read:   readRole,
+		Delete: deleteRole,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"role_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user_ids": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				ForceNew: true,
+			},
+			"permissions": {
+				Type: schema.TypeList,
```

Comment:
With a TypeSet here instead of a TypeList you might overcome the issues you mentioned.